modules/core/mod_macro.c

```diff
@@ -483,6 +483,11 @@ static const char *process_content(apr_pool_t * pool,
     for (i = 0; i < contents->nelts; i++) {
         const char *errmsg;
         /* copy the line and substitute macro parameters */
+	if (strlen(((char**)contents->elts)[i]) >= MAX_STRING_LEN) {
```

Comment:
Shouldn't this be `(strlen(line) + strlen(...)) >= MAX_STRING_LEN` ?

Reply:
```
char line[MAX_STRING_LEN];
apr_cpystrn(line, ((char **) contents->elts)[i], MAX_STRING_LEN);
```
We copy `contents->elts[i]` onto `line`, so we need to be sure that `contents->elts[i]` is shorter than `sizeof(line)`, so `MAX_STRING_LEN`. Did I miss something?

Btw, we could use `sizeof(line)` instead of `MAX_STRING_LEN` everywhere to be cleaner ...

Reply:
Sorry, yes I wasn't read that properly. Makes sense, thanks.

Reply:
Also: yes using `sizeof line` would be better I agree.

Reply:
done